src/libs/SidebarUtils.ts

```diff
@@ -151,7 +151,23 @@ function getOrderedReportIDs(
     const isInDefaultMode = !isInGSDMode;
     const allReportsDictValues = Object.values(allReports);
     // Filter out all the reports that shouldn't be displayed
-    const reportsToDisplay = allReportsDictValues.filter((report) => ReportUtils.shouldReportBeInOptionList(report, currentReportId ?? '', isInGSDMode, betas, policies, true));
+    const reportsToDisplay = allReportsDictValues.filter((report) => {
+        const parentReportActionsKey = `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report?.parentReportID}`;
+        const parentReportActions = allReportActions?.[parentReportActionsKey];
+        const parentReportAction = parentReportActions?.find((action) => action && report && action?.reportActionID === report?.parentReportActionID);
+        const doesReportTransactionThreadHaveViolations = Boolean(
+            betas.includes(CONST.BETAS.VIOLATIONS) && parentReportAction && ReportUtils.doesTransactionThreadHaveViolations({report, transactionViolations, parentReportAction}),
+        );
```

Comment:
Let's cast only `parentReportAction` to boolean, since the other values are boolean, this is a suggestion to show what I mean, but it's probably too long for a single line
```suggestion
        const doesReportTransactionThreadHaveViolations = betas.includes(CONST.BETAS.VIOLATIONS) && Boolean(parentReportAction) && ReportUtils.doesTransactionThreadHaveViolations({report, transactionViolations, parentReportAction},
        );
```

Reply:
This causes a type error on the second use of parentReportAction because it prevents the compiler from narrowing the type. 

ordinarily I would just throw a `!` on the second instance to assert that it isn't null, but such things are forbidden by the linter rules in this project.

Reply:
>This causes a type error on the second use of parentReportAction because it prevents the compiler from narrowing the type.

Can you expand on this? And does it happen if we pass individual params to doesTransactionThreadHaveViolations?

If we have to cast the whole expression, let's add a comment explaining why

Reply:
Got it to work by replacing `Boolean()` with `!!` -- looked it up and:

> You can always coerce values to booleans by running them through the Boolean function, or by using the shorter double-Boolean negation. (The latter has the advantage that TypeScript infers a narrow literal boolean type true, while inferring the first as type boolean.)

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#truthiness-narrowing

Reply:
Basically `Boolean(parentReportAction)` evaluates to `true` so there is no information for the compiler about the type of the parameter (which could be `any`).